app/services/record_declarations/base.rb

```diff
@@ -1,12 +1,37 @@
 # frozen_string_literal: true
 
-require "json_schema/validate_body_against_schema"
-
 module RecordDeclarations
   class Base
-    attr_accessor :params
+    include ActiveModel::Model
+
+    attr_accessor :course_identifier, :user_id, :raw_event, :lead_provider_from_token, :declaration_date, :declaration_type, :evidence_held
+
+    validates :course_identifier, inclusion: { in: :valid_courses_for_user, message: "The property '#/course_identifier' must be an available course to '#/participant_id'" }
```

Comment:
I18n?

Reply:
I hate i18n with a passion, but fine, if you want it a lot I can put it in there.